src/config/mod.rs

```diff
@@ -13,31 +13,29 @@ use snafu::{ResultExt, Snafu};
 use std::fs::DirBuilder;
 use std::path::PathBuf;
 
+mod builder;
+mod compiler;
 pub mod component;
 mod diff;
 mod loading;
+mod unit_test;
 mod validation;
 mod vars;
 pub mod watcher;
 
+pub use builder::ConfigBuilder;
 pub use diff::ConfigDiff;
 pub use loading::{load_from_paths, load_from_str, process_paths, CONFIG_PATHS};
-pub use validation::check;
+pub use unit_test::build_unit_tests_main as build_unit_tests;
+pub use validation::warnings;
 
-#[derive(Deserialize, Serialize, Debug)]
-#[serde(deny_unknown_fields)]
+#[derive(Debug, Default)]
 pub struct Config {
-    #[serde(flatten)]
     pub global: GlobalOptions,
```

Comment:
Should we make those private now?

Reply:
I did try putting most of these behind methods at least, but it ended up being a little more than I want to bite off right now. That's something we can do as a followup.